### PR TITLE
Hyperparameter optimization chapter

### PR DESCRIPTION
**Pull request by mehradans92**
This chapter discusses techniques for hyperparameter optimization in deep learning:

- [x]  Discussion on training hyperparameters (learning rate, number of hidden layers,  number of nodes, dropout rate, batch size and number of epochs to train, regularization, dropout)

- [x] Discussion on hyperparameter optimization techniques (search algorithms and trial schedulers, Bayesian optimization)
- [x] Include an example for hyperparameter tuning on an existing example in the book.

### CODE REVIEW

**Comment by mehradans92** [comment]
@whitead which existing example sounds good to you for the third task?

**Comment by whitead** [comment]
@whitead maybe std layer chapter - which had some discussion of hyperparameter choices

**Comment by whitead** [comment]
@mehradans92 I think you missed some conflicts. Check the `bib` file for `=======`.
